chia/rpc/wallet_rpc_client.py

```diff
@@ -1662,94 +1670,56 @@ async def dao_adjust_filter_level(self, wallet_id: int, filter_level: int) -> di
 
     async def vc_mint(
         self,
-        did_id: bytes32,
+        request: VCMint,
         tx_config: TXConfig,
-        target_address: Optional[bytes32] = None,
-        fee: uint64 = uint64(0),
         extra_conditions: tuple[Condition, ...] = tuple(),
         timelock_info: ConditionValidTimes = ConditionValidTimes(),
-        push: bool = True,
     ) -> VCMintResponse:
-        response = await self.fetch(
-            "vc_mint",
-            {
-                "did_id": encode_puzzle_hash(did_id, "rpc"),
-                "target_address": encode_puzzle_hash(target_address, "rpc") if target_address is not None else None,
-                "fee": fee,
-                "extra_conditions": conditions_to_json_dicts(extra_conditions),
-                "push": push,
-                **tx_config.to_json_dict(),
-                **timelock_info.to_json_dict(),
-            },
+        return VCMintResponse.from_json_dict(
+            await self.fetch(
+                "vc_mint", {**request.json_serialize_for_transport(tx_config, extra_conditions, timelock_info)}
```

Comment:
And similar several times below.  Just suggesting once in case there's a reason to keep it this way.

```suggestion
                "vc_mint", request.json_serialize_for_transport(tx_config, extra_conditions, timelock_info)
```

Reply:
addressed